src/main/scala/util/ClockDivider.scala

```diff
@@ -0,0 +1,41 @@
+package util
+
+import Chisel._
+
+/** Divide the clock by 2 */
+class ClockDivider2 extends Module {
+  val io = new Bundle {
+    val reset_out = Bool(OUTPUT)
+    val clock_out = Clock(OUTPUT)
+  }
+
+  val clock_reg = Reg(Bool())
+  clock_reg := !clock_reg
+
+  // Hold reset_out high until positive edge of clock_out
+  val reset_reg = RegEnable(Bool(false), Bool(true), clock_reg)
```

Comment:
Not sure I really get what reset_out is doing here... at any rate it is not the same as what the verilog model was doing. Could you get the same functionality by just putting this on the last flop in the chain?